src/classad/classad/value.h

```diff
@@ -103,6 +108,26 @@ class Value
 			valueType 	= UNDEFINED_VALUE;
 		}
 
+#ifdef TJ_PICKLE
+		// validate and skip over the next expression in the stream if it is a valid value
+		// returns the number of bytes read from the stream, or 0 on failure.
+		static unsigned int Scan(ExprStream & stm, unsigned char & kind);
+		// get the string value from an ExprStream if the next token is a string value
+		static bool Get(ExprStream & stm, std::string_view & str);
+		static bool Get(ExprStream & stm, std::string & str) {
+			std::string_view sv;
+			int ret = Get(stm, sv);
+			str = sv;
```

Comment:
I tend to expect output parameters to be unmolested in the case of failure, and that generally seems to be true for the other Get() functions.  Consider only assigning to `str` to `ret` is true.

Reply:
agreed. I will change it